src/containers/Society.jsx

```diff
@@ -1,64 +1,124 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
 
 import Page from './Page';
 import PageHeader from '../components/header/PageHeader';
 import ActivityCard from '../components/activities/ActivityCard';
 import MasonryLayout from '../containers/MasonryLayout';
 import Stats from '../components/sidebar/Stats';
-import activities from '../fixtures/activities';
 
 /**
  * @name Society
  * @summary Renders a society page
  * @return React node that displays a society page
  */
-const Society = () => (
-  <Page>
-    <div className='mainContent'>
-      <div className='society'>
-        <PageHeader title='Activities' />
-        <div className='activities'>
-          <MasonryLayout
-            items={
-              activities.map(activity => (
-                <ActivityCard
-                  id={activity.id}
-                  category={activity.category}
-                  date={(activity.date)}
-                  description={activity.activity}
-                  points={activity.points}
-                  status={activity.status}
-                />
-              ))
-            }
-          />
+class Society extends Component {
+  static propTypes = {
+    societyInfo: PropTypes.shape({
+      requesting: PropTypes.bool.isRequired,
+      error: PropTypes.shape({}).isRequired,
+      info: PropTypes.shape({
+        name: PropTypes.string.isRequired,
+        totalPoints: PropTypes.number.isRequired,
+        usedPoints: PropTypes.number.isRequired,
+        remainingPoints: PropTypes.number.isRequired,
+        image: PropTypes.string.isRequired,
+        loggedActivities: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
+      }).isRequired,
+    }).isRequired,
+  };
+
+  /**
+   * React component lifecycle method getDerivedStateFromProps
+   * @param {Object} nextProps - received props
+  */
+  static getDerivedStateFromProps(nextProps) {
+    return {
+      ...nextProps.societyInfo.info,
+    };
+  }
+
+  /**
+   * Society component class constructor
+   * @param {*} props - society information
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      ...props.societyInfo.info,
+    };
+  }
+
+  /**
+   * Render Society Component
+   * @return {object} JSX for Society Component
+   */
+  render() {
+    const showUserDetails = true;
+    const {
+      name,
+      totalPoints,
+      usedPoints,
+      remainingPoints,
+      loggedActivities,
+    } = this.state;
+    return (
+      <Page>
+        <div className='mainContent'>
+          <div className='society'>
+            <PageHeader title='Activities' />
+            <div className='activities'>
+              <MasonryLayout
```

Comment:
@mkiterian How is the `key error` warning on the masonary layout page going to be resolved?  Any ideas?

Reply:
@mkiterian You could use array index for this.

Reply:
Which `key error` is this @jmnyarega? because in the current layout structure, the columns are css-generated hence do not need a key, and the activity card keys are the activity IDs